File: Queue/Enumeration/1212-sequential-digits/sequential-digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        ArrayList<Integer> list = new ArrayList<>();
        Deque<Integer> q = new ArrayDeque<>();
        for(int i=1 ;i<9;i++){
            q.addLast(i);
        }
        while(!q.isEmpty()){
            int num = q.removeFirst();
            if(num >= low){
                list.add(num);
            }
            int digit = num%10;
            num = num*10 + ++digit;
            if(digit <= 9 && num <= high) q.addLast(num);
        }
        return list;
    }
}

File: Queue/Enumeration/1212-sequential-digits/sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        res = []
        q = [1,2,3,4,5,6,7,8,9]
        while q :
            num = q.pop(0)
            if num >= low:
                res.append(num)
            last = num%10
            num = num*10 +(last+1)
            if last < 9 and num <= high:
                q.append(num)
        return res